splipy/utils/__init__.py

```diff
@@ -166,10 +166,76 @@ def uniquify(iterator):
         seen.add(i)
         yield i
 
+def raise_order_1D(n, k, T, P, m, periodic):
+    """ Implementation of method in "Efficient Degree Elevation and Knot Insertion
+        for B-spline Curves using Derivatives" by Qi-Xing Huang a Shi-Min Hu, Ralph R Martin. Only the case of open knot vector is fully implemented
+    :param int n: (n+1) is the number of initial basis functions
+    :param int k: spline order
+    :param T: knot vector
+    :param P: weighted NURBS coefficients
+    :param int m: number of degree elevations
+    :param int periodic: Number of continuous derivatives at start and end. -1 is not periodic, 0 is continuous, etc.
+    :return Q: new control points
+    """
+
+    u = np.unique(T[k-1:-k+1])
```

Comment:
use `u = b.knot_spans()` (see line 187), as this is also sensitive to floating point tolerances

Reply:
`ghost_nodes=False` should take care of `[k-1:k+1]`